### PR TITLE
Add CIDR block support

### PR DESCRIPTION
**Pull request by utkarsh867**
@pheiduck Re-submitting a PR with all the changes for adding CIDR block support (Related to #888).

Still hoping to reproduce #965, but I noticed that every time wg-easy starts, config was not refreshing `address` and `cidrBlock`.

### CODE REVIEW

**Comment by pheiduck** [comment]
Thanks but it looks like it is a bit different to the `master` branch eg. README. Maybe you missed a rebase?

**Comment by pheiduck** [comment]
And seams there is nothing different from the previous main PR I guess this need some rework and should not break the userspace like before.

**Comment by utkarsh867** [comment]
Seems like I had missed some changes in the `README`

**Comment by pheiduck** [comment]
Tahnks, because of the concerns I have already mentioned. This needs enough testing before it is ready to merge into master.
We will merge it first into test branch.

**Comment by Q0** [comment]
I'm planning to take CIDR for testing, I need this function

**Comment by utkarsh867** [comment]
> I'm planning to take CIDR for testing, I need this function

Appreciate it

File: src/lib/WireGuard.js [hunk 1]
```diff
@@ -36,22 +40,32 @@ module.exports = class WireGuard {
...
           config = await fs.readFile(path.join(WG_PATH, 'wg0.json'), 'utf8');
           config = JSON.parse(config);
+          config = {
+            ...config,
+            server: {
+              ...config.server,
+              address,
+              cidrBlock,
+            },
+          };
```
[the remark this answers is not in the document]
Comment (pheiduck) [reply src/lib/WireGuard.js:55]: Ya, needs to be tested. Let me know when we should merge this into test branch.

**Comment by utkarsh867** [comment]
@Q0 did you take this up for testing? Maybe I can go ahead and deploy this on my side as well to test.

**Comment by pheiduck** [comment]
This PR might need some work due it was only a PR copy.

**Comment by pheiduck** [comment]
@utkarsh867 please resolve Conflicting files after that this will be merged.
When @peterlewis approve.

**Comment by pheiduck** [comment]
please fixup install of uuid with: `npm install uuid`

**Comment by pheiduck** [comment]
@utkarsh867 please rebase.

**Comment by utkarsh867** [comment]
sorry @pheiduck I've been quite busy with my work in the past couple of weeks so I have not been able to get this on my schedule. I'll try to rebase this very soon.

Also waiting to see if someone can verify if these changes work well.

**Comment by kaaax0815** [comment]
Will be reimplemented after https://github.com/wg-easy/wg-easy/pull/1333.
The old solution (which introduced breaking changes) will be used for a cleaner solution

**Pull request closed without merging**